### PR TITLE
add candidateAssignedToOthers flag to TaskInfoQuery interface

### PR DESCRIPTION
**Pull request by mlippens**
For discussion see on flowable forum: https://forum.flowable.org/t/taskquery-api-questions-multiple-assignees-and-candidates-that-have-been-assigned/131/6

If needed I am of course willing to adapt this pull request.

git commit message:This extra flag has been implemented in order to allow querying without assignee logic. When setting the flag when querying, one can still request tasks or historic tasks regardless of the set assignee.The behaviour without this flag is that a task that is assigned can never be returned when searching for candidateUsers.

### CODE REVIEW

File: modules/flowable-task-service/src/main/resources/org/flowable/task/db/mapping/entity/Task.xml [hunk 1]
```diff
@@ -802,23 +804,31 @@
           <!-- if dbIdentityUsed set true in process engine configuration -->
           <if test="userIdForCandidateAndAssignee != null">
             <if test="candidateGroups == null">
-              and (RES.ASSIGNEE_ = #{userIdForCandidateAndAssignee} or (RES.ASSIGNEE_ is null and (I.USER_ID_ = #{userIdForCandidateAndAssignee}
-              or I.GROUP_ID_ IN (select g.GROUP_ID_ from ${prefix}ACT_ID_MEMBERSHIP g where g.USER_ID_ = #{userIdForCandidateAndAssignee} ) ) ) )
+              and (RES.ASSIGNEE_ = #{userIdForCandidateAndAssignee}
+                or (
+                <if test="candidateAssignedToOthers">
```
Comment (tijsrademakers) [inline modules/flowable-task-service/src/main/resources/org/flowable/task/db/mapping/entity/Task.xml]: Should this be <if test="!candidateAssignedToOthers"> instead?

**Comment by tijsrademakers** [comment]
Thanks, added a comment about an if check, could you validate if my assumption is correct?

File: modules/flowable-task-service/src/main/resources/org/flowable/task/db/mapping/entity/Task.xml (hunk 1 above)
Comment (mlippens) [reply modules/flowable-task-service/src/main/resources/org/flowable/task/db/mapping/entity/Task.xml]: I assume you meant the missing negation? You are correct, good catch, I updated it.
It seems that that branch of the query is never tested (neither in the new tests nor in the original ones), hence why it slipped through.

**Comment by tijsrademakers** [comment]
Merged the PR manually, thanks.
I've changed the boolean to ignoreAssigneeValue, hope that's ok for you.

**Comment by mlippens** [comment]
@tijsrademakers that's quite alright.

**Pull request closed without merging**